src/kagglehub/gcs_upload.py

```diff
@@ -81,23 +83,33 @@ def _upload_blob(file_path: str, model_type: str):
 
 
 def upload_files(folder: str, model_type: str, quiet: bool = False):  # noqa: FBT002, FBT001
-    """upload files in a folder. Throws an exception if more than 50 files are found.
+    """upload files in a folder. Zips the files if there are more than 50.
     Parameters
     ==========
     folder: the folder to upload from
     quiet: suppress verbose output (default is False)
     model_type: Type of the model that is being uploaded.
     """
-    # TODO(b/312511716): Handle case where more than 50 files are to be uploaded by first zipping the content
 
     # Count the total number of files
     file_count = 0
     for _, _, files in os.walk(folder):
         file_count += len(files)
 
     if file_count > MAX_FILES_TO_UPLOAD:
-        max_files_to_upload_exception = "Cannot upload more than 50 files. Consider zipping the files first."
-        raise ValueError(max_files_to_upload_exception)
+        if not quiet:
+            logger.info(f"More than {MAX_FILES_TO_UPLOAD} files detected, creating a zip archive...")
+
+        with TemporaryDirectory() as temp_dir:
+            zip_path = os.path.join(temp_dir, "archive.zip")
```

Comment:
move `archive.zip` to a const at the top of this file and reuse it in `test_model_upload.py`.